src/Build/Graph/ProjectGraph.cs

```diff
@@ -573,6 +530,20 @@ private static ImmutableList<string> ExpandDefaultTargets(ImmutableList<string>
                         .InsertRange(i, defaultTargets);
                     i += defaultTargets.Count;
                 }
+                else if (targets[i].Equals(MSBuildConstants.ProjectReferenceTargetsOrDefaultTargetsMarker, StringComparison.OrdinalIgnoreCase))
+                {
+                    var targetsString = graphEdge.GetMetadataValue(ItemMetadataNames.ProjectReferenceTargetsMetadataName);
+
+                    var expandedTargets = string.IsNullOrEmpty(targetsString)
+                        ? defaultTargets
+                        : ExpressionShredder.SplitSemiColonSeparatedList(targetsString).ToList();
+
+                    targets = targets
+                        .RemoveAt(i)
+                        .InsertRange(i, expandedTargets);
+
+                    i += defaultTargets.Count;
```

Comment:
`defaultTargets` or `expandedTargets`?

Reply:
Good catch! Fortunately the defect does not affect correctness.